### DISCUSSION TITLE
GitHub Pages Not Deploying

### DISCUSSION THREAD

**Discussion by angeidheal** [community/community#23885] [Other Feature Feedback, Questions, & Ideas]
<p>I pushed some changes to <a href="https://geidh.uk/" rel="nofollow noopener">my site</a> around 2 hours ago. The changes have appeared in <a href="https://github.com/GeidhUK/GeidhUK.github.io" rel="nofollow noopener">my repo</a> but have not appeared on the actual website.</p>
<p>As a test, I’ve since tried adding a sentence to the very bottom of the <a href="https://macmhicheil.uk/mu-dheidhinn/" rel="nofollow noopener">about page</a> of another website I have hosted on GitHub pages and again the change appears in <a href="https://github.com/MacMhicheil/macmhicheil.github.io/blob/master/mu-dheidhinn.md" rel="nofollow noopener">my repo</a> but does not appear on the website.</p>
<p>I’ve tried CTRL+SHIFT+R to refresh, incognito mode, different browsers and different machines but no joy.</p>
<p>Both sites are being built from the master-branch (and always have been) and in fact neither has any other branches so I know it is not that.</p>
<p>Any other ideas what could be wrong?</p>

**Comment by Burd89** [comment]
<p>I seem to be having a similar issue. It throws a 404 error when accessing <a href="https://burd89.github.io/Test3/" rel="nofollow noopener">https://burd89.github.io/Test3/</a>.</p>
<p>My github experience is limited, but it was working a few hours ago. However, whatever I’m trying to do now does not seem to affect the outcome.</p>
<p>Few things I have tried:</p>
<ul>
<li>Added a blank space to index.html</li>
<li>changed the name of the repo</li>
<li>waited for an hour</li>
<li>Redeployed</li>
</ul>
<p>Also, the environment page does not show a recent deployment. I would expect, if I publish a page, then there would be a recent deployment.</p>

**Comment by ericworks** [comment, marked as answer]
<p>Same here. I found no page build in activity log under <code>environment</code>. It seems like a server issue, cause the Github Page has a <code>degraded performance</code> now. Current server status at <a href="https://www.githubstatus.com/" rel="nofollow noopener">https://www.githubstatus.com/</a> .</p>
<p>A temp workaround:  disable the Github Page and then enable it again in repo <code>setting</code>. The page should be forced to rebuild. </p>
<hr>
<p>update: don’t try disable the Github Page since the new build could last forever. And your page will report 404 during this time!</p>

**Reply by MUZDOG1** [reply]
is anybody else having the same issue regarding deployments not even starting yet my changes are showed in my repo? really need some help because im so lost.

**Comment by cmuething** [comment]
<p>I am having the same issue.</p>
<p>Reached out to support but have not heard back.</p>
<p>Looks like a server issue.</p>

**Comment by angeidheal** [comment]
<p>Thanks, I’ve just seen the ‘degraded performance’ message on the Status page. It didn’t show anything earlier when I checked but agree this is probably the cause.</p>

**Comment by KrisGray** [comment]
<p>Now that the servers have recovered, I checked to see if my github pages site was built. It wasn’t so I had to make a very small change within the guthub source editor and committed straight to master. This restarted the build and now my site has been updated.</p>

**Comment by Tom-Okafor** [comment]
<p>please can you help me my site isnt publishing. I’m getting a 404 not found error. please help… Thanks</p>

**Comment by LJBarnes** [comment]
<p>Making a small change and committing to master did the trick for me as well. Thanks for sharing your solution!</p>

**Comment by kylemhall** [comment]
<p>Having the same issue right now, even though others have reported it working two hours ago.</p>

**Comment by glyn** [comment]
<p>I’m also having the same issue. Disabling and re-enabling github pages in the repository settings doesn’t seem to have any effect. Forcing a change of git SHA at head, e.g. by changing the commit log, may have done the trick, although there was a delay, so I’m wondering if there was an outage of some sort (although the github status page is showing status Normal for Github Pages). I pushed a change to the web page text a few minutes ago and again no deployment has spun up. Could it be that this is a busy time and that there is a long queue?</p>

**Comment by ppoon23** [comment]
<p>Same issue here. I pushed some changes 10hrs ago and worked fine. However, Github pages has not been deploying for the past 30mins.</p>

**Comment by brpetrucci** [comment]
<p>Same here. Yesterday it was updating fine, today it stopped. Tried incognito mode, refreshed caches, changed publishing branch and changed back etc, nothing works.</p>

**Comment by czhang2718** [comment]
<p>I have tried every suggestion I found online (empty commit, different branch, hard refresh, cache), but my page stopped deploying, after multiple updates, an hour ago. However, according to <a href="https://www.githubstatus.com/" rel="noopener nofollow ugc">https://www.githubstatus.com/</a> github pages is working fine. Is anyone else’s <a href="http://github.io" rel="noopener nofollow ugc">github.io</a> page not updating?</p>
<p>Edit: Nevermind, I think it’s just extremely slow.</p>

**Comment by TimDaub** [comment]
<p>Hey, it seems I’m running into this problem now. Over on my <a href="https://github.com/TimDaub/timdaub.github.io/deployments/activity_log?environment=github-pages" rel="noopener nofollow ugc">blog</a>, commits have stopped deploying even after committing multiple times.</p>
<p>The GitHub status says that everything is green though. So what’s happening?</p>

**Comment by DaintyGames** [comment]
<p>My last commits haven’t been deployed in about 30 minutes</p>

**Comment by jonathanl-telenav** [comment]
<p>I am having the same problem right now (2021.09.06-12.34pm PT)</p>

**Comment by b0bbins** [comment]
<p>I’m also having this problem, sounds like there’s an issue with Pages that isn’t being surfaced in server status… disappointing</p>

**Comment by lsli8888** [comment]
<p>Same issue here with GitHub Pages not deploying.</p>

**Comment by jonathanl-telenav** [comment]
<p>It’s back up again. My pages are publishing.</p>

**Comment by jonathanl-telenav** [comment]
<p>Actually, it’s really bogged down is all. My new edit is taking a long time.</p>

**Comment by gabrielarpino** [comment]
My github pages not deploying. Tried all the recommendations above and still nothing.

**Reply by gabrielarpino** [reply]
Solution I found: sign up for github pro again. Github Pages stopped deploying once my pro expired, without warning.

**Reply by mervess** [reply]
This was the case for me as well. Turns out _private github pages repo_ only works with the **pro** accounts. In case anyone else's page is not deployed for the same reason, I'm leaving it here.

**Reply by dhimmel** [reply]
Ah there really should be a better error message or banner saying deployments have been stopped!

**Reply by TheBlueFriend** [reply]
Yeah it's very annoying. They don't have a documentation written anywhere about it. I only got to know from my friend. Was stressed out for days.

**Reply by liaocs2008** [reply]
Thanks! It turns out turning visibility on and off cannot keep repo private while being deployable on github...

**Comment by ghost** [comment]
it happened again. but they are looking into it

**Reply by vivian1223** [reply]
Same here :(

**Reply by JulioPDX** [reply]
+1

**Reply by bkpleb** [reply]
Experiencing rn. Sad :( 
I really wanted to see some very important changes immediately.

**Comment by ranbo** [comment]
I was having the same problem: I'd check in code, it would say it was deployed, and if I clicked "View Deployment", it was still code from a couple weeks ago. I tried checking in a new little change, and about the same time, checked in another little change to a new branch and created a pull request to merge in. One of those things apparently worked, because suddenly it was updated.

**Comment by ianmaloba** [comment]
Me having the same issue in 2023 :(

**Comment by MSoup** [comment]
Same issue here. No idea what to do or what the workaround is.

**Comment by h0lybyte** [comment]
xD its 2023, yall still slacking with the github pages! Its so confusing, when it could be so easy to use.

**Comment by mervess** [comment]
Guys I'm leaving it here in case it fits to somebody's case: 
If your GitHub pages repo is private, and your account is a free (and no pro) account, then it does not deploy. 
The same thing also happens when you downgrade your account. It should be public in such cases to be deployed.

**Reply by gcantoni** [reply]
Can confirm.

**Comment by judithlk** [comment]
I experienced the same thing, but I realised it's because my project is a react app and I hadn't updated the build. In case you're working with react too, check out the link below for a solution.

https://github.com/Frnt-End/Update-React-in-GitHub-Pages

**Comment by marsgazer** [comment]
If there is an error in your project, github pages do not deploy your site and do not tell you what the error is, and even do not tell you that there is an error. It looks like this is a policy: you don't debug on the production site.

**Comment by ipsitapriyadarsini** [comment]
I have deployed my application it was working, but after new push the deployed URL is showing like https://ipsitapriyadarsini.github.io/portfolio/ this.
![image](https://github.com/community/community/assets/41966804/0723bedc-875c-4caf-9e10-0300f6521025)

**Comment by sainipankaj15** [comment]
@marsgazer , I don't think it is like that, I have tried my code in local. It is working fine.
But again same issue not working for me :(

What I think , this thing takes time to happened. I am deploying after 8 months till now i have done so many deployment , it happened within 1 or 2 minutes but this time it is happening

**Comment by noararazzak** [comment]
Anyone having issues with "page build and deployment". I tried four different times in the last half hour and it failed every time. It happened once last week, around May 13 but the issue resolved itself.

**Reply by mbh038** [reply]
I am having this issue. Everything was working fine up until yesterday (30 May 2024), then I added a couple of .Rmd files and now today no rebuilds get deployed.  I cannot understand why.

**Reply by noararazzak** [reply]
I faced the issue when I edited one of the .html files in my website but when I reverted it back, the website compiled. If the website still compiles if reverted to an earlier version, it could be some coding error that stops the website from being compiled by Github pages. 

I am new to html programming and maintain only one Github page, so do not have any concrete solution.

**Reply by mbh038** [reply]
I have tried stripping the repo back to its bare essentials, but no joy. I cannot even deploy the minimal example you get from `bookdown:::bookdown_skeleton(getwd())`. Each time I try I get a message from GitHub telling me that some jobs were not successful in page build and run.

**Comment by kelewinska** [comment]
Is anyone experiencing any issues with page deployment? I tried to update my personal [page ](https://kelewinska.github.io/)a few days ago, and despite the successfully pushed commit, it never went online. Today I reversed the version to the last working one, and introduced edits again from scratch, just in case there were any errors in the html code, and again the page is not being deployed.
I am aware that a few days ago there was some downtime for deployment functionality due to an issue with workbooks, but the issue has been resolved since. Were there any recent changes (past 2 weeks) in the protocols or anything that might affect the personal pages?

**Reply by kelewinska** [reply]
OK, it turned out that there has been a change in 'Actions' and now I need to allow GitHub to build the webpage. All sorted out.

**Comment by mbh038** [comment]
What do you have to do in Actions? I cannot get my pages to deploy either.

**Reply by kelewinska** [reply]
you need to allow other users than yourself to have action permissions. 
it is done by enabling the 4th permission option in the general action section related to the repository of your website. then you need to push the commits anew and the page should be deployed within a few minutes.
![image](https://github.com/user-attachments/assets/cb36114b-344a-405f-9460-ceda7ba1a916)

**Reply by mbh038** [reply]
Thanks very much . Now it deploys quickly but I just get 404 File Not Found page. Another thing to get to the bottom of.....

**Comment by Kalcitus** [comment]
I also had this issue, but it had been a while since I last made changes (almost a year). It seems, in that time, github began requiring you to have a ".nojekyll" file in the root if you're not using Jekyll. After I added this, my changes started getting published again and I stopped getting 404 on new pages.

**Reply by MUZDOG1** [reply]
im trying this right now, thanks. ps, are there any other solutions? ive been experiencing this issue for a while now.

**Comment by MUZDOG1** [comment]
does anybody have any other solutions or working solutions to this issue? im really struggling to find a solution that works, writing on 5/1/2025

**Comment by goberlan** [comment]
How is there no good answer for this? The big problem is there isn't even a way to check a log or something to see why it might be failing. Which is insane.

**Reply by goberlan** [reply]
Because I didn't see it mentioned, you can use `gh api` to get the status of your build pages:
https://docs.github.com/en/rest/pages/pages?apiVersion=2022-11-28#get-latest-pages-build (found at this stack link: https://stackoverflow.com/a/51544901)
```
gh api \
  -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  /repos/OWNER/REPO/pages/builds/latest 
```
and I got this in return:
```
  "url": "https://github.company.com/api/v3/repos/owner/repo/pages/builds/42253",
  "status": "errored",
  "error": {
    "message": "Site contained a symlink that should be dereferenced: /stable. For more information, see https://docs.github.com/enterprise-server@3.16/github/working-with-github-pages/troubleshooting-jekyll-build-errors-for-github-pages-sites#config-file-error."
  },
```
So the reason it didn't build is because it didn't like the symlink. Would be pretty nice if this showed up in github in the actual `Pages` location...

**Reply by goberlan** [reply]
Which lead me to see this:
https://github.com/orgs/community/discussions/9104#discussioncomment-5208620

github no longer supports symlinks for non-Action builds... I'm sure there is a good reason.

**Comment by jurri** [comment]
You have to wait... how long dont know... github write ~10min.
last push I waited ... 
Kill the Repo are instantly done ... but create new one takes time to show online...

[look here from github docs](https://docs.github.com/en/pages/quickstart#creating-your-website)
Visit username.github.io to view your new website. Note that **it can take up to 10 minutes for changes to your site to publish after you push** the changes to GitHub.
